wasmtime-wasi/src/host_impls.rs

```diff
@@ -92,6 +92,33 @@ pub fn wasmtime_ssp_proc_exit(rval: wasm32::__wasi_exitcode_t) {
     ::std::process::exit(rval as i32)
 }
 
+pub unsafe fn wasmtime_ssp_fd_prestat_get(
+    prestats: *mut host::fd_prestats,
+    fd: host::__wasi_fd_t,
+    buf: *mut host::__wasi_prestat_t,
+) -> host::__wasi_errno_t {
+    host::rwlock_rdlock(&mut (*prestats).lock as *mut host::rwlock);
+
+    let mut prestat: *mut host::fd_prestat = ::std::ptr::null_mut();
+    let error = host::fd_prestats_get_entry(prestats, fd, &mut prestat);
+
+    if error != host::__WASI_ESUCCESS {
+        host::rwlock_unlock(&mut (*prestats).lock as *mut host::rwlock);
+        return error;
+    }
+
+    (*buf).pr_type = host::__WASI_PREOPENTYPE_DIR as host::__wasi_preopentype_t;
```

Comment:
It's the `host::__wasi_errno_t` nightmare again, that is, type mismatch: `host::__WASI_PREOPENTYPE_DIR == u32` while it should be `host::__wasi_preopentype_t == u8`. It might be a good opportunity to manually define all `__wasi_preopentype_t` values. What do you think?

Reply:
I think you're right. And similar to the errno situation, bringing in these defines into host.rs is where we'll need to go eventually anyway.

Reply:
Coolios. I'll include it in this PR then :-)